app/experimenter/jetstream/client.py

```diff
@@ -152,33 +157,40 @@ def get_experiment_data(experiment):
     }
 
     for window in windows:
-        data = raw_data[window] = JetstreamData(
-            __root__=(get_data(recipe_slug, window) or [])
-        )
-        result_metrics, primary_metrics_set, other_metrics = get_results_metrics_map(
-            data,
-            experiment.primary_outcomes,
-            experiment.secondary_outcomes,
-            outcomes_metadata,
-        )
-
-        if data and window == "overall":
-            # Append some values onto Jetstream data
-            data.append_population_percentages()
-            data.append_retention_data(raw_data["weekly"])
-
-            ResultsObjectModel = create_results_object_model(data)
-            data = ResultsObjectModel(result_metrics, data, experiment)
-            data.append_conversion_count(primary_metrics_set)
-
-            experiment_data["other_metrics"] = other_metrics
-        elif data and window == "weekly":
-            ResultsObjectModel = create_results_object_model(data)
-            data = ResultsObjectModel(result_metrics, data, experiment, window)
-
-        transformed_data = data.dict(exclude_none=True) or None
-
-        experiment_data[window] = transformed_data
+        experiment_data[window] = {}
+        data_from_jetstream = get_data(recipe_slug, window) or []
+        segments = list(set(point["segment"] for point in data_from_jetstream))
+
+        for segment in segments:
+            segment_data = [d for d in data_from_jetstream if d["segment"] == segment]
```

Comment:
Not that it really matters, the scale of this data isn't really anything to worry about, but just a fun little optimization to think about.  

Here we're looping over the points in `data_from_jetstream` twice, once to get the unique set of segments, and then once again to get the points of each segment type.  You could do that in one pass like

```py
segment_points = defaultdict(list)
for point in data_from_jetstream:
    segment_points[point['segment']].append(point)

for segment, segment_data in segment_points.items():
    ...
```

Reply:
Perfect, just the kind of easy optimization I like!